### PR TITLE
SWC-7799

### PR DESCRIPTION
**Pull request by jay-hodgson**

### CODE REVIEW

File: packages/synapse-react-client/src/assets/icons/CloudWarning.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import React from 'react'
+import SvgIcon, { SvgIconProps } from '@mui/material/SvgIcon'
```
Comment (jay-hodgson) [inline packages/synapse-react-client/src/assets/icons/CloudWarning.tsx:2]: New CloudWarning icon

File: packages/synapse-react-client/src/components/DownloadCart/DownloadIneligibleForPackagingFilesFromListButton.module.scss [hunk 2]
```diff
@@ -0,0 +1,154 @@
+.externalWarningPanel {
```
Comment (jay-hodgson) [inline packages/synapse-react-client/src/components/DownloadCart/DownloadIneligibleForPackagingFilesFromListButton.module.scss:16]: Refactored all styles to scss module

File: packages/synapse-react-client/src/components/DownloadCart/DownloadIneligibleForPackagingFilesFromListButton.tsx [hunk 3]
```diff
@@ -22,8 +33,9 @@ import {
 } from '@/synapse-queries/download/useDownloadList'
 import { getFiles } from '@/synapse-client/SynapseClient'
 import { useGetEntityQueryOptions } from '@/synapse-queries/entity/useEntity'
-import { DialogBase } from '../DialogBase'
```
Comment (jay-hodgson) [inline packages/synapse-react-client/src/components/DownloadCart/DownloadIneligibleForPackagingFilesFromListButton.tsx:25]: Now built from fixed position Paper component rather than the DialogBase

File: packages/synapse-react-client/src/components/DownloadCart/DownloadIneligibleForPackagingFilesFromListButton.tsx [hunk 4]
```diff
@@ -49,6 +61,10 @@ function supportsFileSystemAccess(
 }
 
 const FILE_BATCH_SIZE = 50 // Maximum batch size for fetching presigned URLs
+const ETA_DELAY_MS = 3000 // Wait this long before showing ETA
```
Comment (jay-hodgson) [inline packages/synapse-react-client/src/components/DownloadCart/DownloadIneligibleForPackagingFilesFromListButton.tsx:64]: Give us some time to download before reporting an estimated time for download

File: packages/synapse-react-client/src/components/DownloadCart/DownloadIneligibleForPackagingFilesFromListButton.tsx [hunk 5]
```diff
@@ -299,6 +321,7 @@ export function DownloadIneligibleForPackagingFilesFromListButton(
               // For external file handles, use the externalURL
               downloadUrl = (fileResult.fileHandle as ExternalFileHandle)
                 .externalURL
+              setShowExternalFileWarning(true)
```
Comment (jay-hodgson) [inline packages/synapse-react-client/src/components/DownloadCart/DownloadIneligibleForPackagingFilesFromListButton.tsx:363]: This is the new Paper component "attached" above the Download Progress Paper component.

File: packages/synapse-react-client/src/components/DownloadCart/DownloadIneligibleForPackagingFilesFromListButton.tsx [hunk 6]
```diff
@@ -380,6 +405,7 @@ export function DownloadIneligibleForPackagingFilesFromListButton(
               if (response.body) {
                 const reader = response.body.getReader()
                 let bytesReceived = 0
+                let streamAborted = false
```
Comment (jay-hodgson) [inline packages/synapse-react-client/src/components/DownloadCart/DownloadIneligibleForPackagingFilesFromListButton.tsx:450]: Bug fix.  Noticed that cancelling the download didn't actually cancel the current download (it only stopped future batches from being processed).  Now it checks on every chunk

**Comment by jay-hodgson** [comment]
Screenshot 2026-05-06 at 5 14 54 PM

-------
Without the external file warning (either dismissed, or never encountered an external file handle):
Screenshot 2026-05-06 at 5 15 02 PM


Note - also tested with a very large file name, and verified it looks ok in the dialog (overflow ellipsis)

File: packages/synapse-react-client/src/components/DownloadCart/DownloadIneligibleForPackagingFilesFromListButton.tsx [hunk 7]
```diff
@@ -389,21 +415,59 @@ export function DownloadIneligibleForPackagingFilesFromListButton(
...
 
+                    // Check for cancellation on every chunk
+                    if (isCancelledRef.current) {
+                      streamAborted = true
+                      await writableStream.abort()
+                      break
+                    }
+
+                    if (fileDownloadStartRef.current === null) {
+                      fileDownloadStartRef.current = Date.now()
```
Comment (jay-hodgson) [inline packages/synapse-react-client/src/components/DownloadCart/DownloadIneligibleForPackagingFilesFromListButton.tsx:469]: Start the timer when the first chunk is received.   Calculate elapsed time, and estimated seconds remaining based on the current transfer rate and size of the file

File: packages/synapse-react-client/src/components/DownloadCart/DownloadIneligibleForPackagingFilesFromListButton.tsx [hunk 8]
```diff
@@ -389,21 +415,59 @@ export function DownloadIneligibleForPackagingFilesFromListButton(
                     bytesReceived += value.length
                     await writableStream.write(value)
 
+                    // Check for cancellation on every chunk
+                    if (isCancelledRef.current) {
+                      streamAborted = true
+                      await writableStream.abort()
+                      break
+                    }
```
[the remark this answers is not in the document]
Comment (jay-hodgson) [reply packages/synapse-react-client/src/components/DownloadCart/DownloadIneligibleForPackagingFilesFromListButton.tsx:466]: Fixed

File: packages/synapse-react-client/src/components/DownloadCart/DownloadIneligibleForPackagingFilesFromListButton.tsx [hunk 9]
```diff
@@ -683,68 +747,96 @@ export function DownloadIneligibleForPackagingFilesFromListButton(
...
+                <Typography className={styles.progressStatsText}>
+                  {downloadProgress.estimatedSecondsRemaining !== null
+                    ? `About ${dayjs
+                        .duration({
+                          seconds: downloadProgress.estimatedSecondsRemaining,
+                        })
+                        .humanize()} remaining`
+                    : ''}
+                </Typography>
+              </Box>
```
[the remark this answers is not in the document]
Comment (jay-hodgson) [reply packages/synapse-react-client/src/components/DownloadCart/DownloadIneligibleForPackagingFilesFromListButton.tsx:878]: Fixed (verified with Mac screen reader)

File: packages/synapse-react-client/src/components/DownloadCart/DownloadIneligibleForPackagingFilesFromListButton.module.scss [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+  position: fixed;
+  bottom: calc(150px + 10px); // positioned above the progress panel
+  right: 20px;
+  z-index: 1300; // MUI modal z-index
```
[the remark this answers is not in the document]
Comment (jay-hodgson) [reply packages/synapse-react-client/src/components/DownloadCart/DownloadIneligibleForPackagingFilesFromListButton.module.scss]: Fixed

File: packages/synapse-react-client/src/components/DownloadCart/DownloadIneligibleForPackagingFilesFromListButton.module.scss [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+  margin: -1px;
+  overflow: hidden;
+  padding: 0;
+  position: absolute;
+  width: 1px;
+}
+
+.externalWarningPanel {
+  position: fixed;
+  bottom: calc(150px + 10px); // positioned above the progress panel
```
Comment (hallieswan) [inline packages/synapse-react-client/src/components/DownloadCart/DownloadIneligibleForPackagingFilesFromListButton.module.scss]: Could consider moving the progress panel height into a constant, so this calculation is self-documenting

File: packages/synapse-react-client/src/components/DownloadCart/DownloadIneligibleForPackagingFilesFromListButton.tsx [hunk 12]
```diff
@@ -389,21 +457,60 @@ export function DownloadIneligibleForPackagingFilesFromListButton(
...
+                      reader.cancel()
+                      await writableStream.abort()
+                      break
+                    }
+
+                    if (fileDownloadStartRef.current === null) {
+                      fileDownloadStartRef.current = Date.now()
+                    }
+                    const elapsed =
+                      (Date.now() - fileDownloadStartRef.current) / 1000
```
Comment (hallieswan) [inline packages/synapse-react-client/src/components/DownloadCart/DownloadIneligibleForPackagingFilesFromListButton.tsx]: Nit: what do you think about keeping `elapsed` in milliseconds, rather than converting to seconds? That would allow us to simplify the comparison with `ETA_DELAY_MS`. Could also consider including the units in the variable name for clarity, e.g. `elapsedMs`

File: packages/synapse-react-client/src/components/DownloadCart/DownloadIneligibleForPackagingFilesFromListButton.test.tsx [hunk 13]
```diff
@@ -177,25 +295,159 @@ describe('DownloadIneligibleForPackagingFilesFromListButton', () => {
     expect(mockShowDirectoryPicker).not.toHaveBeenCalled()
   })
 
-  // Note: Full integration testing of the download flow with File System Access API
```
Comment (hallieswan) [inline packages/synapse-react-client/src/components/DownloadCart/DownloadIneligibleForPackagingFilesFromListButton.test.tsx:180]: It looks like some of these scenarios are still untested -- would it be worth updating the comment based on what's currently tested rather than deleting it entirely?

**Review by hallieswan — APPROVED** [review]
Looks great! Just a few optional suggestions

**Review by hallieswan — APPROVED** [review]
